inference-engine/src/offline_transformations/include/pruning.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/pass/graph_rewrite.hpp>
+#include <ngraph/pass/manager.hpp>
+
+namespace ngraph {
+namespace pass {
+
+class InitConstMask;
+class PropagateMasks;
+class ShrinkWeights;
+
+class Pruning;
+
+} // namespace pass
+} // namespace ngraph
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief Check Constant operation values by given dimensions and set
+ * masks according to results that are bases on `condition` lambda function.
+ * Works for Constant with floating point type (f16, f32, f64).
+ */
+class ngraph::pass::InitConstMask : public ngraph::pass::MatcherPass {
```

Comment:
I propose to avoid such definition like `class ngraph::...::ClassName` because doxygen has some problems with such definition.
I already have fixed all such places inside the nGraph core, let's avoid such code style.

Reply:
Mask can be shared between ops that doesn't change it.